catalog/confluence/catalog-confluence-source/src/main/java/org/codice/ddf/confluence/source/ConfluenceInputTransformer.java

```diff
@@ -0,0 +1,333 @@
+        metacard.setAttribute(Topic.CATEGORY, confluenceType);
+        metacard.setAttribute(Topic.VOCABULARY, CONFLUENCE_TYPE_LINK);
+
+        metacard.setTitle(getString(json, "title"));
+
+        metacard.setAttribute(Confluence.JSON_RESPONSE, json.toString());
+
+    }
+
+    private void parseLabels(MetacardImpl metacard, Object json)
+            throws CatalogTransformerException {
+        Object labelsElement = getJsonElement(json, "metadata", "labels");
+        if (labelsElement != null) {
+            ArrayList<String> labels = new ArrayList<>();
+            getJsonArray(labelsElement, "results").stream()
+                    .forEach(e -> labels.add(getStringOrDefault(e, UNKNOWN, "name")));
+            metacard.setAttribute(Topic.KEYWORD, labels);
+        }
+    }
+
+    private void parseHistory(MetacardImpl metacard, Object json)
+            throws CatalogTransformerException {
+        Object history = getRequiredJsonElement(json, "history");
+
+        metacard.setModifiedDate(getDate(getRequiredJsonElement(history, "lastUpdated", "when")));
+
+        metacard.setCreatedDate(getDate(getRequiredJsonElement(history, "createdDate")));
+
+        metacard.setAttribute(Contact.CREATOR_NAME, getString(history, "createdBy", "username"));
+
+        ArrayList<String> contributors = new ArrayList<>();
+        getJsonArray(history, "contributors", "publishers", "users").stream()
+                .forEach(e -> addContributor(contributors, e));
+        metacard.setAttribute(Contact.CONTRIBUTOR_NAME, contributors);
+
+    }
+
+    private void parseLinks(MetacardImpl metacard, Object json, String baseUrl)
+            throws CatalogTransformerException {
+        if (baseUrl == null) {
+            baseUrl = getStringOrDefault(json, "", "baseUrl");
+        }
+
+        Object links = getRequiredJsonElement(json, "_links");
+
+        String downloadUrl = getStringOrDefault(links, null, "download");
+        if (downloadUrl != null) {
+            metacard.setAttribute(Metacard.RESOURCE_DOWNLOAD_URL,
+                    String.format("%s%s", baseUrl, downloadUrl));
+            metacard.setAttribute(Metacard.RESOURCE_URI,
+                    String.format("%s%s", baseUrl, downloadUrl));
+            String fileSize = getStringOrDefault(json, null, "extensions", "fileSize");
+            if (fileSize != null) {
+                metacard.setAttribute(Metacard.RESOURCE_SIZE, fileSize);
+            }
+        }
+        ArrayList<String> associations = new ArrayList<>();
+        if (StringUtils.isNotEmpty(baseUrl)) {
+            associations.add(String.format("%s%s", baseUrl, getString(links, "webui")));
+            associations.add(baseUrl);
+        }
+
+        String space = getStringOrDefault(json, null, "space", "_links", "webui");
+        if (space != null) {
+            associations.add(baseUrl + space);
+        }
+        metacard.setAttribute(Associations.EXTERNAL, associations);
```

Comment:
Is there value to assigning an empty list to an Attribute? Null attributes are culled, but this would remain present, yet empty, if both the conditionals were not met.

I doubt this would cause any issues, but would it be worth an !associations.isEmpty() check before setting?